MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm

```diff
@@ -701,6 +985,44 @@
     }
 }
 
+void MVKImageDescriptor::encodeToMetalArgumentBuffer(MVKResourcesCommandEncoderState* rezEncState,
+													 id<MTLArgumentEncoder> mtlArgEncoder,
+													 uint32_t descSetIndex,
+													 MVKDescriptorSetLayoutBinding* mvkDSLBind,
+													 uint32_t elementIndex,
+													 MVKShaderStage stage,
+													 bool encodeToArgBuffer,
+													 bool encodeUsage) {
+	VkDescriptorType descType = getDescriptorType();
+	uint8_t planeCount = (_mvkImageView) ? _mvkImageView->getPlaneCount() : 1;
+
+	for (uint8_t planeIndex = 0; planeIndex < planeCount; planeIndex++) {
+		uint32_t planeDescIdx = (elementIndex * planeCount) + planeIndex;
+
+		id<MTLTexture> mtlTexture = _mvkImageView ? _mvkImageView->getMTLTexture(planeIndex) : nil;
+		if (encodeToArgBuffer) {
+			uint32_t argIdx = mvkDSLBind->getMetalResourceIndexOffsets().textureIndex + planeDescIdx;
+			[mtlArgEncoder setTexture: mtlTexture atIndex: argIdx];
```

Comment:
I did too. But CTS seems to be holding up so far.

I originally included code to extract the Metal texture type from the shader conversion reflection and populate the `MTLArgumentDescriptor` to match. This would mean separate encoders per pipeline.

We'll still have to pull info from the pipeline in the case where the encoders and arg buffers do not behave homogeneously, as in iOS or earlier macOS. Currently, the plan is to go all in there, and pull the encoder from the `MTLFunction` itself, in an attempt to get perfect alignment between the encoder and the shader. This will mean one encoder/arg buffer per shader stage. I've started the plumbing for that in this PR, but it requires substantial expansion in future releases.